chat/message/user.go

```diff
@@ -71,6 +73,34 @@ func (u *User) LastMsg() time.Time {
 	return u.lastMsg
 }
 
+// SetAway sets the users availablity state
+func (u *User) SetAway(msg string) {
+	u.mu.Lock()
+	defer u.mu.Unlock()
+	u.awayStatus = msg
+}
+
+// SetActive sets the users as active
+func (u *User) SetActive() {
+	u.mu.Lock()
+	defer u.mu.Unlock()
+	u.awayStatus = ""
+}
+
+// GetAway returns if the user is away, when they went away and if they set a message
+func (u *User) GetAway() (bool, time.Time, string) {
+	u.mu.Lock()
+	defer u.mu.Unlock()
+	if u.awayStatus == "" {
+		return false, time.Time{}, ""
+	}
+	lastPublicActivity := u.lastMsg
+	if u.lastMsg.IsZero() {
+		lastPublicActivity = u.joined
+	}
+	return true, lastPublicActivity, u.awayStatus
```

Comment:
GetAway should return the time when away was set, not the time since the last activity. You'll need a separate timestamp, like `awayTime time.Time`.

Reply:
updated in next commit!